### PR TITLE
Fix GTest_SOURCE=BUNDLED not work

### PR DESCRIPTION
**Pull request by JackyWoo**
### Description

I failed to build velox in ubuntu and clang-15, because I have gtest-1.11.0 installed (to successully build we need v1.13.0).
When I try to set `-DGTest_SOURCE=BUNDLED` the cmake output shows that it does not work. 

```
/data1/home/wujianchao/.cache/JetBrains/RemoteDev/dist/1729cdbf0ed8b_CLion-2024.2.2/bin/cmake/linux/x64/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=/usr/lib/llvm-15/bin/clang -DCMAKE_CXX_COMPILER=/usr/lib/llvm-15/bin/clang++ -DGTest_SOURCE=BUNDLED -DCMAKE_PREFIX_PATH=/data1/home/wujianchao/project/jd/velox/deps-install -G "Unix Makefiles" -S /data1/home/wujianchao/project/jd/velox -B /data1/home/wujianchao/project/jd/velox/cmake-build-debug

....

-- Could NOT find double-conversion (missing: double-conversion_DIR)
-- Found double-conversion: /usr/lib/x86_64-linux-gnu/libdouble-conversion.so (Required is at least version "3.1.5")
-- Found GTest: /usr/local/lib/cmake/GTest/GTestConfig.cmake (found version "1.11.0")
-- Using SYSTEM GTest

````

### CODE REVIEW

**Review by assignUser — APPROVED** [review]
Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -571,7 +571,7 @@ include_directories(SYSTEM velox/external)
 
 # these were previously vendored in third-party/
 if(NOT VELOX_DISABLE_GOOGLETEST)
-  set(GTest_SOURCE AUTO)
+  set_source(GTest)
   resolve_dependency(GTest)
```
Comment (assignUser) [inline CMakeLists.txt:575]: Actually, if we have a minimum version I wonder why it's not set here @majetideepak @kgpai ? https://github.com/facebookincubator/velox/blob/main/CMake/resolve_dependency_modules/README.md doesn't seem to list it though
```suggestion
  resolve_dependency(GTest 1.13)
```

Comment (majetideepak) [reply CMakeLists.txt:575]: We have no minimum versions for gtest and glog. We depend on the default from the platform/system package manager.

Comment (JackyWoo) [reply CMakeLists.txt]: I specified minmimun versionf for gtest and glog.

Comment (assignUser) [reply CMakeLists.txt]: > to successully build we need v1.13.0

What issue do you have? I just checked and our CI containers use 1.11 so I don't think that minimum version is required for velox. 

So in that case please remove the minimum version again unless you can demonstrate that they are needed for current `main` state (if you need it for a PR please handle it there)

Comment (JackyWoo) [reply CMakeLists.txt:575]: I just rollback the minimum versions and I will try to reproduce it (for the environment has changed needs some time)

**Comment by JackyWoo** [comment]
The `Linux Build `test failed is because `VELOX_DEPENDENCY_SOURCE` is set to `SYSTEM` but there is no gtest in docker. 

Install gtest or set `VELOX_DEPENDENCY_SOURCE` to auto which do you prefer? Maybe we should fix it in another PR.

**Comment by assignUser** [comment]
> Install gtest or use bundled gtest which do you prefer?

I think we can use bundled for now and add gtest to the setup scripts in a follow up PR :)

**Pull request closed without merging**